### PR TITLE
Bring the repo up to date

### PR DESCRIPTION
**Pull request by anvacaru**
**What's changed**
 - README.md file updated with up-to-date kontrol instructions
 - removed the old `src/KEVMCheats.sol` and included the new Kontrol library for cheat codes https://github.com/runtimeverification/kontrol-cheatcodes
 - removed a duplicated IERC20.sol interface
 - updated `doit`, `exclude` and `erc20.sh` scripts
 - Toy example `TokenTest.testTransfer` proof is now passing.

### CODE REVIEW

**Comment by RaoulSchaffranek** [comment]
Should we rename this repo to `kontrol-demo`?

File: README.md [hunk 1]
```diff
@@ -23,39 +24,41 @@ Then, run `foundryup` in a new terminal session or after reloading your `PATH`.
...
+- the [`erc20.sh`](./erc20.sh) file contains a script that runs all the ERC20 proofs and times them.
 
 ### Contracts
 
 In the [`src`](./src) subdirectory, you will find multiple files:
 
 - `ERC20.sol`: The file contains the base ERC20 token Solidity contract from [Open Zeppelin](https://github.com/OpenZeppelin/openzeppelin-contracts/tree/master/contracts/token/ERC20).
 - `IERC20Metadata.sol`, `IERC20.sol` and `Context.sol` are helper files of the ERC20 contract, imported from the same repository.
-- `KEVMCheats.sol`: The [KEVMCheats.sol](./src/utils/KEVMCheats.sol) contract interface contains functions which are only available to KEVM.
+- `KEVMCheats.sol`: The [KEVMCheats.sol](./src/utils/KEVMCheats.sol) contract interface contains functions which are only available to `kontrol`.
```
Comment (lisandrasilva) [inline README.md:61]: I think this contract no longer exists, so the link does not work.

File: doit [hunk 2]
```diff
@@ -1,86 +1,53 @@
...
-}
-
-do_kompile() {
-    kevm kompile --pyk --backend haskell ${verification_file} --definition ${definition} \
-        --main-module VERIFICATION --syntax-module VERIFICATION
-}
-
-do_prove() {
-    kevm prove ${spec_file} --definition ${definition} --pyk ${verbose} --save-directory ${kcfgs_dir}
+    kevm foundry-section-edge ${verbose} ${test} ${edge_id} ${bug_report} "$@"
```
Comment (lisandrasilva) [inline doit:48]: Isn't it `kontrol section-edge`?

Comment (anvacaru) [reply doit]: Missed this one. Yes, you are correct!

**Comment by anvacaru** [comment]
> Should we rename this repo to `kontrol-demo`?

@RaoulSchaffranek Good idea! I don't have the rights to change the repo name. Perhaps @ehildenb ?

**Pull request merged**